### PR TITLE
Fix saving customer with required fields

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | See issue.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/13556
| How to test?  | Adding/editing customer should check for required fields now.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13887)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks I've fixed this issue, however, I don't think it should behave like that. For FO customers it makes sense to have required fields, but as BO admin, I should be able to add customer without required fields or disable "Partner offers" even though it's required in FO.

File: src/Adapter/Customer/CommandHandler/EditCustomerHandler.php [hunk 1]
```diff
@@ -77,6 +79,24 @@ public function handle(EditCustomerCommand $command)
 
         $this->updateCustomerWithCommandData($customer, $command);
 
+        // validateFieldsRequiredDatabase() below is using $_POST
+        // to check if required fields are set
+        $_POST[RequiredField::PARTNER_OFFERS] = $command->isPartnerOffersSubscribed();
```
Comment (PierreRambaud) [inline src/Adapter/Customer/CommandHandler/EditCustomerHandler.php:83]: duplicate code from AddCustomerHandler. Must be refactor.

**Comment by sarjon** [comment]
@matks could you rerun build?

**Comment by PierreRambaud** [comment]
@sarjon Done

**Comment by sarjon** [comment]
Perfect, all green! :green_heart:

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
> @matks I've fixed this issue, however, I don't think it should behave like that. For FO customers it makes sense to have required fields, but as BO admin, I should be able to add customer without required fields or disable "Partner offers" even though it's required in FO.

This can be relevant indeed. I can think of usecases where it makes sense and other when it does not make sense. So let's merge the PR but we can raise the question for a global Bo improvement.

Ping @colinegin

**Comment by colinegin** [comment]
Thanks for raising the question ! To be honest, I don't really understand why you should not make mandatory fields also mandatory when you create a customer from the BO. This will create 2 types of customers with different rules...

**Comment by sarjon** [comment]
> This will create 2 types of customers with different rules...

No, I don't think so. :thinking: 

There is a simple use case for that.

Let's say "Partner offers subscription" is required and in FO customer registers with it. Then customer contacts you to remove him/her from subscription list and as a merchant/admin, you cannot do that. 

The workaround is to disable that field globally (this one https://prnt.sc/nry3hz so we are on the same page), update customer and then enable required field globally again. 

For me, it looks like a wrong behavior.

**Comment by matks** [comment]
Thank you @sarjon

**Pull request merged**